diff -r 484d3f8e5978 sys/src/cmd/rio/dat.h
--- a/sys/src/cmd/rio/dat.h	Tue Sep 29 07:28:47 2020 -0700
+++ b/sys/src/cmd/rio/dat.h	Sat Oct 10 15:16:44 2020 +0300
@@ -344,8 +344,11 @@
 char		*startdir;
 int		sweeping;
 int		wctlfd;
+int		capsfd;
+Channel		*capsc;
 char		srvpipe[];
 char		srvwctl[];
+char		srvcaps[];
 int		errorshouldabort;
 int		menuing;		/* menu action is pending; waiting for window to be indicated */
 int		snarfversion;	/* updated each time it is written */
diff -r 484d3f8e5978 sys/src/cmd/rio/fsys.c
--- a/sys/src/cmd/rio/fsys.c	Tue Sep 29 07:28:47 2020 -0700
+++ b/sys/src/cmd/rio/fsys.c	Sat Oct 10 15:16:44 2020 +0300
@@ -50,6 +50,7 @@
 
 char	srvpipe[64];
 char	srvwctl[64];
+char	srvcaps[64];
 
 static	Xfid*	filsysflush(Filsys*, Xfid*, Fid*);
 static	Xfid*	filsysversion(Filsys*, Xfid*, Fid*);
@@ -115,6 +116,9 @@
 	return 0;
 }
 
+void
+capsproc(void *v);
+
 Filsys*
 filsysinit(Channel *cxfidalloc)
 {
@@ -152,6 +156,12 @@
 	post(srvwctl, "wctl", p0);
 	close(p0);
 
+	if(cexecpipe(&p0, &capsfd) < 0)
+		goto Rescue;
+	snprint(srvcaps, sizeof(srvcaps), "/srv/riocaps.%s.%d", fs->user, pid);
+	post(srvcaps, "caps", p0);
+	close(p0);
+
 	/*
 	 * Start server processes
 	 */
@@ -160,6 +170,12 @@
 		error("wctl channel");
 	proccreate(wctlproc, c, 4096);
 	threadcreate(wctlthread, c, 4096);
+
+	capsc = chancreate(sizeof(char*), 0);
+	if(capsc == nil)
+		error("caps channel");
+	proccreate(capsproc, capsc, 4096);
+
 	proccreate(filsysproc, fs, 10000);
 
 	/*
@@ -724,3 +740,31 @@
 	d.mtime = clock;
 	return convD2M(&d, buf, nbuf);
 }
+
+void
+capsproc(void *v)
+{
+	char *s;
+	int n, eofs;
+	Channel *c;
+
+	threadsetname("CAPSPROC");
+	c = v;
+
+	eofs = 0;
+	for(;;){
+		if((s = recvp(c)) == nil)
+			break;
+		n = write(capsfd, s, strlen(s));	/* room for \0 */
+		free(s);
+		if(n < 0)
+			break;
+		if(n == 0){
+			if(++eofs > 20)
+				break;
+			continue;
+		}
+		eofs = 0;
+	}
+	close(capsfd);
+}
diff -r 484d3f8e5978 sys/src/cmd/rio/rio.c
--- a/sys/src/cmd/rio/rio.c	Tue Sep 29 07:28:47 2020 -0700
+++ b/sys/src/cmd/rio/rio.c	Sat Oct 10 15:16:44 2020 +0300
@@ -1,4 +1,5 @@
 #include <u.h>
+#include <stdio.h>
 #include <libc.h>
 #include <draw.h>
 #include <thread.h>
@@ -344,9 +345,15 @@
 	threadsetname("keyboardthread");
 
 	while(s = recvp(kbdchan)){
-		if(*s == 'k' || *s == 'K')
+		if(*s == 'k' || *s == 'K') {
 			shiftdown = utfrune(s+1, Kshift) != nil;
-		if(input == nil || sendp(input->ck, s) <= 0)
+		}
+		if (capsc != nil && !shiftdown && strlen(s) == 4 && utfrune(s+1, Kcaps) != nil) {
+			char *caps = strdup("caps\n");
+			if (sendp(capsc, caps) <= 0)
+				free(caps);
+			free(s);
+		} else if(input == nil || sendp(input->ck, s) <= 0)
 			free(s);
 	}
 }
